patches/StartOfRoundPatch.cs

```diff
@@ -0,0 +1,118 @@
+                }
+                continue;
+            }
+            AudioLowPassFilter component = currentVoiceChatAudioSource.GetComponent<AudioLowPassFilter>();
+            OccludeAudio component2 = currentVoiceChatAudioSource.GetComponent<OccludeAudio>();
+            component.enabled = true;
+            component2.overridingLowPass = flag || ___allPlayerScripts[i].voiceMuffledByEnemy;
+            currentVoiceChatAudioSource.GetComponent<AudioHighPassFilter>().enabled = flag;
+            if (!flag)
+            {
+                currentVoiceChatAudioSource.spatialBlend = 1f;
+                playerControllerB2.currentVoiceChatIngameSettings.set2D = false;
+                currentVoiceChatAudioSource.bypassListenerEffects = false;
+                currentVoiceChatAudioSource.bypassEffects = false;
+                currentVoiceChatAudioSource.outputAudioMixerGroup = SoundManager.Instance.playerVoiceMixers[playerControllerB2.playerClientId];
+                component.lowpassResonanceQ = 1f;
+            }
+            else
+            {
+                currentVoiceChatAudioSource.spatialBlend = 0f;
+                playerControllerB2.currentVoiceChatIngameSettings.set2D = true;
+                if (GameNetworkManager.Instance.localPlayerController.isPlayerDead)
+                {
+                    currentVoiceChatAudioSource.panStereo = 0f;
+                    currentVoiceChatAudioSource.outputAudioMixerGroup = SoundManager.Instance.playerVoiceMixers[playerControllerB2.playerClientId];
+                    currentVoiceChatAudioSource.bypassListenerEffects = false;
+                    currentVoiceChatAudioSource.bypassEffects = false;
+                }
+                else
+                {
+                    if (playerControllerB.currentlyHeldObjectServer is WalkieTalkie) {
+                        currentVoiceChatAudioSource.panStereo = 0.4f;
+                        Debug.Log("Playing through right ear");
+                    } 
+                    else {
+                        currentVoiceChatAudioSource.panStereo = 0f;
+                        Debug.Log("Playing normally");
+                    }
+                    currentVoiceChatAudioSource.bypassListenerEffects = false;
+                    currentVoiceChatAudioSource.bypassEffects = false;
+                    currentVoiceChatAudioSource.outputAudioMixerGroup = SoundManager.Instance.playerVoiceMixers[playerControllerB2.playerClientId];
+                }
+                component2.lowPassOverride = 4000f;
+                component.lowpassResonanceQ = 3f;
+            }
+            if (GameNetworkManager.Instance.localPlayerController.isPlayerDead)
+            {
+                playerControllerB2.voicePlayerState.Volume = 0.8f;
+            }
+            else
+            {
+                playerControllerB2.voicePlayerState.Volume = 1f;
+            }
+        }
+
+        return false; // Skip the original method
+    }
+}
+
```

Comment:
As mentioned above I think it would be easiest to review and test if we broke the configuration changes into a separate PR from these spatial audio adjustments. I have a few thoughts/questions here as well:

1. I've read through this but haven't grokked what has changed (don't have the time to pull up the decompiler and compare at the moment). It looks to me very similar to the original function so I assume you made a relatively small change and have copied the majority of the original function over. Can you give me a description of what is changed?
2. I'm hoping with this mod to avoid overriding vanilla functions as much as possible. This should prevent breakage and regression if the original dev makes subtle logic updates to those functions. I'm curious to know what the issue was and how this corrects it, we may be able to implement the equivalent in a less destructive way.